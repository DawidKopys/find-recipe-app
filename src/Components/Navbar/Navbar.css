.recipe-finder-app__navbar {
  margin: 0 25px;
  margin-top: 10px;
  height: 45px;
}

/* Navbar main layout */
.navbar {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

/* Navbar overlay */
.navbar::before {
  display: none;
  content: '';
  z-index: -1;
  position: fixed;
  top: 0;
  right: 0;
  width: 100vw;
  height: 100vh;
  background-color: rgba(0, 0, 0, 0);
}

.navbar--with-overlay::before {
  display: block;
  z-index: 3;
  background-color: rgba(0, 0, 0, 0.2);
}

/* Links sidebar */
.navbar__links-list {
  padding: 20px;
  padding-top: 60px;
  z-index: 4;
  position: fixed;
  top: 0;
  right: 0;
  width: 70vw;
  height: 100vh;
  background-color: var(--color-grey-1);
  transform: translateX(100%);
  transition: none;
}

.navbar__links-list--open {
  transform: translateX(0);
  transition: transform 0.2s ease-in;
}

.navbar__list-item {
  height: 45px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.navbar__link {
  position: relative;
}

.navbar__link--active {
  box-shadow: -17px 0 0 var(--color-grey-1), -20px 0 0 var(--color-purple);
}

/* Close Sidebar Button */
.navbar__btn-close-sidebar {
  position: absolute;
  width: 25px;
  height: 25px;
  top: 25px;
  right: 25px;
  background-image: url('Assets/icon-cross-grey.svg');
}

/* Sign In and Sign Out Buttons */
.navbar__list-item--sign-in,
.navbar__list-item--sign-up {
  margin-top: 15px;
}

.navbar__link--sign-in,
.navbar__link--sign-up {
  width: 100%;
  padding: 9px 0;
  border-radius: 5px;
  box-shadow: 1px 1px 4px rgba(0, 0, 0, 0.25);
}

.navbar__link--sign-in {
  border: 2px solid var(--color-purple);
  color: var(--color-purple);
  background-color: white;
}

.navbar__link--sign-up {
  border: 2px solid var(--color-purple);
  background-color: var(--color-purple);
  color: white;
}

/* Typography */
.navbar__links-list {
  font-family: 'Roboto';
  font-size: 1.125rem;
  font-weight: 700;
}

.navbar__link {
  text-align: center;
}

/* Logo Typography */
.reciply-logo {
  font-family: 'Secular One';
  font-size: 24px;
  font-weight: 400;
  text-transform: uppercase;
}

.reciply-logo__first-letter {
  font-family: 'Sansation';
  font-size: 32px;
  font-weight: 700;
  color: var(--color-purple);
  text-transform: uppercase;
}

@media screen and (min-width: 800px) {
  .recipe-finder-app__navbar {
    height: unset;
    grid-area: navbar;
    margin-right: 45px;
  }

  /* Links section */
  .navbar__links-list {
    position: unset;
    transform: unset;
    transition: none;
    padding: 0;
    width: unset;
    height: unset;
    background-color: unset;
    display: flex;
    align-self: stretch;
    align-items: stretch;
    justify-content: flex-end;
  }

  .navbar__list-item {
    margin-left: 30px;
  }

  /* Active page link underline */
  .navbar__link--active {
    box-shadow: unset;
  }

  .navbar__link--active::after {
    content: '';
    display: block;
    height: 2px;
    width: 100%;
    border-radius: 1px;
    background-color: var(--color-purple);
    position: absolute;
    top: 28px;
  }

  /* Sign In and Sign Out Buttons */
  .navbar__list-item--sign-in,
  .navbar__list-item--sign-up {
    margin-top: 0;
  }

  .navbar__link--sign-in,
  .navbar__link--sign-up {
    width: 100px;
  }

  /* Close and Open Burger Menu Button */
  .navbar__btn-open-sidebar,
  .navbar__btn-close-sidebar {
    display: none;
  }
}
