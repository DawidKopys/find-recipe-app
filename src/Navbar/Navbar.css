@media screen and (min-width: 800px) {
  .recipe-finder-app__navbar {
    grid-area: navbar;
  }
}

.navbar {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.navbar__logo {
  margin-left: 20px;
}

.navbar__links-list {
  height: 100%;
  display: flex;
  align-items: stretch;
  justify-content: flex-end;
}

.navbar__list-item {
  margin-right: 30px;
  display: flex;
  align-items: center;
}

.navbar__link {
  pointer-events: none; /* temp */
  color: black;
}

.navbar__link--sign-in,
.navbar__link--sign-up {
  padding: 9px 11px;
  border-radius: 5px;
  box-shadow: 1px 1px 4px rgba(0, 0, 0, 0.25);
}

.navbar__link--sign-in {
  border: 2px solid var(--color-purple);
  color: var(--color-purple);
}

.navbar__link--sign-up {
  border: 2px solid var(--color-purple);
  background-color: var(--color-purple);
  color: white;
}

/* Typography */
.navbar__links-list {
  font-family: 'Roboto';
  font-size: 1.125rem;
  font-weight: 700;
}
