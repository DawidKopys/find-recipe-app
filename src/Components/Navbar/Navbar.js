import React, { useState, useEffect, useRef } from 'react';
import BurgerMenu from 'Assets/icon-burger-menu-grey.svg';
import './Navbar.css';
import { useOnClickOutside } from 'CustomHooks/useOnClickOutside';
import { Link, NavLink } from 'react-router-dom';

const Navbar = () => {
  const sidebarRef = useRef();
  useOnClickOutside(sidebarRef, () => setIsSidebarOpen(false));
  const [isSidebarOpen, setIsSidebarOpen] = useState(false);

  useEffect(() => {
    if (isSidebarOpen) {
      document.body.style.overflow = 'hidden';
    } else {
      document.body.style.overflow = 'unset';
    }
  }, [isSidebarOpen]);

  return (
    <nav
      className={`navbar${
        isSidebarOpen ? ' navbar--with-overlay' : ''
      } recipe-finder-app__navbar`}
    >
      <span className='reciply-logo'>
        <span className='reciply-logo__first-letter'>R</span>eciply
      </span>
      <button
        className='navbar__btn-open-sidebar btn'
        onClick={() => setIsSidebarOpen(true)}
      >
        <img src={BurgerMenu} alt='Show Menu' />
      </button>
      <ul
        className={`navbar__links-list${
          isSidebarOpen ? ' navbar__links-list--open' : ''
        }`}
        ref={sidebarRef}
      >
        <li className='navbar__list-item'>
          <button
            className='navbar__btn-close-sidebar btn'
            onClick={() => setIsSidebarOpen(false)}
          />
        </li>
        <li className='navbar__list-item'>
          <NavLink
            to='/add-recipe'
            exact
            activeClassName='navbar__link--active'
            className='navbar__link'
          >
            Add Recipe
          </NavLink>
        </li>
        <li className='navbar__list-item'>
          <NavLink
            to='/'
            exact
            activeClassName='navbar__link--active'
            className='navbar__link'
          >
            Find Recipe
          </NavLink>
        </li>
        <li className='navbar__list-item navbar__list-item--sign-in'>
          <Link to='/' className='navbar__link navbar__link--sign-in'>
            Sign In
          </Link>
        </li>
        <li className='navbar__list-item navbar__list-item--sign-up'>
          <Link to='/' className='navbar__link navbar__link--sign-up'>
            Sign Up
          </Link>
        </li>
      </ul>
    </nav>
  );
};

export default Navbar;
