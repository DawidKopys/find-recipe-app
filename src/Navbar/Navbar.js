import React from 'react';
import Logo from 'Assets/logo-reciply-purple-uppercase.svg';
import './Navbar.css';

const Navbar = () => {
  return (
    <nav className='navbar recipe-finder-app__navbar'>
      <img src={Logo} alt='Reciply Logo' className='navbar__logo' />
      <ul className='navbar__links-list'>
        <li className='navbar__list-item'>
          <a href='' className='navbar__link'>
            Add Recipe
          </a>
        </li>
        <li className='navbar__list-item'>
          <a href='' className='navbar__link'>
            Find Recipe
          </a>
        </li>
        <li className='navbar__list-item'>
          <a href='' className='navbar__link navbar__link--sign-in'>
            Sign In
          </a>
        </li>
        <li className='navbar__list-item'>
          <a href='' className='navbar__link navbar__link--sign-up'>
            Sign Up
          </a>
        </li>
      </ul>
    </nav>
  );
};

export default Navbar;
