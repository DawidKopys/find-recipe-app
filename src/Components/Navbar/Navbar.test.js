import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Navbar from './Navbar.js';

describe('Navbar', () => {
  test('snapshot renders', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
