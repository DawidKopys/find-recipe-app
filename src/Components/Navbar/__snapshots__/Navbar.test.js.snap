// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot renders 1`] = `
<nav
  className="navbar recipe-finder-app__navbar"
>
  <span
    className="reciply-logo"
  >
    <span
      className="reciply-logo__first-letter"
    >
      R
    </span>
    eciply
  </span>
  <button
    className="navbar__btn-open-sidebar btn"
    onClick={[Function]}
  >
    <img
      alt="Show Menu"
      src="test-file-stub"
    />
  </button>
  <ul
    className="navbar__links-list"
  >
    <li
      className="navbar__list-item"
    >
      <button
        className="navbar__btn-close-sidebar btn"
        onClick={[Function]}
      />
    </li>
    <li
      className="navbar__list-item"
    >
      <a
        className="navbar__link"
        href="/add-recipe"
        onClick={[Function]}
      >
        Add Recipe
      </a>
    </li>
    <li
      className="navbar__list-item"
    >
      <a
        className="navbar__link"
        href="/"
        onClick={[Function]}
      >
        Find Recipe
      </a>
    </li>
    <li
      className="navbar__list-item navbar__list-item--sign-in"
    >
      <a
        className="navbar__link navbar__link--sign-in"
        href="/"
        onClick={[Function]}
      >
        Sign In
      </a>
    </li>
    <li
      className="navbar__list-item navbar__list-item--sign-up"
    >
      <a
        className="navbar__link navbar__link--sign-up"
        href="/"
        onClick={[Function]}
      >
        Sign Up
      </a>
    </li>
  </ul>
</nav>
`;
